### PR TITLE
Add macros to enable offline Trino compiles and speed up Trino compiles

### PR DESCRIPTION
**Pull request by jkylling**
The macros are used to disable listing of relations from the warehouse. Run `dbt compile` with the flag `--vars 'no-relation-listing: "true"'` on the command line to enable this. This speeds up DBT compile times. It must not be used with `dbt run`.

This enables running `dbt compile` without access to a Trino cluster. It is also faster than running with a Trino cluster.

Potentially we could also use this to make the local setup of `dbt` simpler:
* We add a `profiles.yml` file to the root of spellbook:
```
spellbook:
  target: dev
  outputs:
    dev:
      type: trino
      user: trino
      password: trino
      host: trino
      port: 1234
      database: hive
      schema: dbt
      threads: 1
```
* User setup from a fresh repository is then:
  * `pipenv install`
  * `pipenv shell`
  * `dbt deps`
* Dev workflow: `dbt compile` (we use the profile name instead of a vars to enable or disable listings)

### CODE REVIEW

**Review by 0xRobin — COMMENTED** [review]
This is great!!
We could add:
```yml
vars:
    no-relation-listing: "{{ env_var('DBT_NO_RELATION_LISTING', 'false') }}"
```
to the `dbt_project.yml` file.
This would allow us (wizards) to set it once as an env var and then forget about it.

**Comment by jeff-dude** [comment]
> This is great!! We could add:
> 
> ```yaml
> vars:
>     no-relation-listing: "{{ env_var('DBT_NO_RELATION_LISTING', 'false') }}"
> ```
> 
> to the `dbt_project.yml` file. This would allow us (wizards) to set it once as an env var and then forget about it.

good call, similar to usage [here](https://github.com/duneanalytics/spellbook/blob/main/dbt_project.yml#L11), then we can apply in our local `~/.zshrc` file, such as:
image

**Comment by jkylling** [comment]
I'm wondering if we could take this even further, and enable this for contributors by default.

We could add a `profiles.yml` file to the root of the repository, and have
```
spellbook:
  target: dev
  outputs:
    dev:
      type: trino
      user: trino
      password: trino
      host: trino
      port: 1234
      database: hive
      schema: dbt
      threads: 1
```
and in `dbt_project.yml` keep
```
profile: "spellbook"
```
The macros are modified to have an if statement `{%- if target.profile_name == 'spellbook' -%}`, which enables offline usage. Then contributors will only need to run `dbt compile`, without any extra arguments or configuration setup elsewhere.

Maintainers will need to set up different profiles (e.g., `spellbook-dev`, `spellbook-prod`), and specify this on the command line if they want to execute `dbt run` directly.

**Comment by jeff-dude** [comment]
> I'm wondering if we could take this even further, and enable this for contributors by default.
> 
> We could add a `profiles.yml` file to the root of the repository, and have
> 
> ```
> spellbook:
>   target: dev
>   outputs:
>     dev:
>       type: trino
>       user: trino
>       password: trino
>       host: trino
>       port: 1234
>       database: hive
>       schema: dbt
>       threads: 1
> ```
> 
> and in `dbt_project.yml` keep
> 
> ```
> profile: "spellbook"
> ```
> 
> The macros are modified to have an if statement `{%- if target.profile_name == 'spellbook' -%}`, which enables offline usage. Then contributors will only need to run `dbt compile`, without any extra arguments or configuration setup elsewhere.
> 
> Maintainers will need to set up different profiles (e.g., `spellbook-dev`, `spellbook-prod`), and specify this on the command line if they want to execute `dbt run` directly.

trying to think through those implications. i like it from the contributors perspective, but open questions on maintainers:
- if `main` branch only contains `profile.yml` at the root directory of spellbook as mentioned above, then we as maintainers would keep our profiles locally like we do now under `~/.dbt/profiles.yml` and have two profiles within, dev and prod
- can we somehow automate switching from dev/prod via variables, rather than applying in command line each time?

**Comment by jkylling** [comment]
> 
> trying to think through those implications. i like it from the contributors perspective, but open questions on maintainers:
> 
> * if `main` branch only contains `profile.yml` at the root directory of spellbook as mentioned above, then we as maintainers would keep our profiles locally like we do now under `~/.dbt/profiles.yml` and have two profiles within, dev and prod

Yes, this is correct.

> * can we somehow automate switching from dev/prod via variables, rather than applying in command line each time?

Maybe, what is the workflow you have in mind here? Having to explicitly write `dbt run --profile dev` or `dbt run --profile prod` is a small safeguard against mistakes.

File: README.md [hunk 1]
```diff
@@ -130,36 +130,6 @@ pipenv shell
 
 You have now created a virtual environment for this project. You can read more about virtual environments [here](https://realpython.com/pipenv-guide/).
```
Comment (dot2dotseurat) [inline README.md:131]: the dunesql readme draft was updated here if we want to use that version https://github.com/duneanalytics/spellbook-trino/pull/44

**Comment by dot2dotseurat** [comment]
target `prod` will run commands without the user's table prefix i.e. nft.trades versus dbt_jeff_nft.trades. unless we are actually trying to circumvent dbt cloud to run something to expose in the data explorer, we shouldn't use prod in the profiles.yml. I don't think we are restricted to dev and prod as targets. So we could have a local target which uses the macro and dev target for the `dbt_jeff...` prefixed workflow.

**Comment by jkylling** [comment]
> target `prod` will run commands without the user's table prefix i.e. nft.trades versus dbt_jeff_nft.trades. unless we are actually trying to circumvent dbt cloud to run something to expose in the data explorer, we shouldn't use prod in the profiles.yml. I don't think we are restricted to dev and prod as targets. So we could have a local target which uses the macro and dev target for the `dbt_jeff...` prefixed workflow.

Should we have two targets in `profiles.yml`, `dev` and `prod`? E.g.,
```yaml
spellbook-local:
  target: dev
  outputs:
    dev:
      type: trino
      user: trino
      password: trino
      host: trino
      port: 1234
      database: hive
      schema: dbt
      threads: 1
spellbook-local-prod:
  target: prod
  outputs:
    prod:
      type: trino
      user: trino
      password: trino
      host: trino
      port: 1234
      database: hive
      threads: 1
```

**Comment by dot2dotseurat** [comment]
@jkylling what is the purpose of the prod profile for local dev? I'm not following.

**Comment by jkylling** [comment]
> @jkylling what is the purpose of the prod profile for local dev? I'm not following.

Sorry, I think I misunderstood your comment. In `spellbooks/profiles.yml` we should only have a single profile, `spellbook-local`, the one added in this PR. Dune developers will have custom profiles to access dev and prod directly in `~/.dbt/profiles.yml`.

**Comment by jeff-dude** [comment]
i'm trying to go through the flow locally on this branch, but running into issues on the internal dune team flow.

for awareness, i changed the profile name back to `spellbook`, to help simplify the change locally, as most dev's local profile will have that value too. it still works fine when mimicking the workflow for wizards. i can get compile to run as expected when using the `profiles.yml` in the root directory of spellbook.

important note to include in contribution guide: wizards will have to be in the root directory to run compile. we need to explicitly call that out, as it can be easy to navigate directories and forget. otherwise, the profile won't be found.

as for the dune workflow, i've tried two ways:
- set the `DBT_PROFILES_DIR` env variable to my personal `~/.dbt/profiles.yml`
- remove the env variable, run with profile param: `dbt compile --profiles-dir ~jdude/.dbt/profiles.yml`

when i run `dbt debug --config-dir`, it correctly shows my personal profile yml file. however, i repeatedly get the error:
```
Encountered an error while reading the project:  ERROR: Runtime Error
  Could not find profile named 'spellbook'
21:15:30  Encountered an error:
Runtime Error
  Could not run dbt
```

i've confirmed i'm connected to tailscale. i've also used this profile to run on `spellbook-trino` repo successfully. i don't think it's even getting to authentication, it just can't find the `spellbook` profile, yet i'm sure it's set to that locally.

**Comment by jkylling** [comment]
> i'm trying to go through the flow locally on this branch, but running into issues on the internal dune team flow.
> 
> for awareness, i changed the profile name back to `spellbook`, to help simplify the change locally, as most dev's local profile will have that value too. it still works fine when mimicking the workflow for wizards. i can get compile to run as expected when using the `profiles.yml` in the root directory of spellbook.
> 
> important note to include in contribution guide: wizards will have to be in the root directory to run compile. we need to explicitly call that out, as it can be easy to navigate directories and forget. otherwise, the profile won't be found.
> 
> as for the dune workflow, i've tried two ways:
> 
> * set the `DBT_PROFILES_DIR` env variable to my personal `~/.dbt/profiles.yml`
> * remove the env variable, run with profile param: `dbt compile --profiles-dir ~jdude/.dbt/profiles.yml`
> 
> when i run `dbt debug --config-dir`, it correctly shows my personal profile yml file. however, i repeatedly get the error:
> 
> ```
> Encountered an error while reading the project:  ERROR: Runtime Error
>   Could not find profile named 'spellbook'
> 21:15:30  Encountered an error:
> Runtime Error
>   Could not run dbt
> ```
> 
> i've confirmed i'm connected to tailscale. i've also used this profile to run on `spellbook-trino` repo successfully. i don't think it's even getting to authentication, it just can't find the `spellbook` profile, yet i'm sure it's set to that locally.

For the dune dev flow, could you use the `--profile spellbook-trino` flag, and not setting environment variables or the `--profiles-dir` flag (notice, singular form of profile)? This will look in the usual locations, not find spellbook-trino in /profiles.yml, and proceed to find it in ~/.dbt/profiles.yml.

**Comment by jeff-dude** [comment]
image

from what i see, we confirm that the default profile is used. `dbt compile` works fine. `dbt compile --profile spellbook-trino` still can't find the profile (i changed to this in my local profile).

i guess ultimately we can still merge this during the freeze, then find the gaps for dune team local setup after. as long as compile works for the wizards.

**Review by jeff-dude — APPROVED** [review]
wizard workflow:
`dbt compile` -- leverages the `profiles.yml` setup in root of spellbook, no connection to trino. note: user must be in root of spellbook on command line to run.

dune workflow:
`dbt compile --profiles-dir ~/.dbt --profile spellbook-trino` -- requires profile setup locally, with correct profile name within

**Comment by jeff-dude** [comment]
> wizard workflow: `dbt compile` -- leverages the `profiles.yml` setup in root of spellbook, no connection to trino. note: user must be in root of spellbook on command line to run.
> 
> dune workflow: `dbt compile --profiles-dir ~/.dbt --profile spellbook-trino` -- requires profile setup locally, with correct profile name within

fyi @Hosuke @0xRobin @couralex6 @a-monteiro @dot2dotseurat 
this will have impact on your local setup when in the spellbook repo

**Pull request merged**